platform/rct/hal/usart_stm32/usart_device_stm32.c

```diff
@@ -181,22 +179,34 @@ static void enableRxDMA(uint32_t RCC_AHBPeriph,
 static void enableRxTxIrq(USART_TypeDef * USARTx, uint8_t usartIrq,
                           uint8_t IRQ_priority, uart_irq_type_t irqType)
 {
-    NVIC_InitTypeDef NVIC_InitStructure;
-
-    /* Configure the NVIC Preemption Priority Bits */
-    NVIC_PriorityGroupConfig(NVIC_PriorityGroup_4);
-
-    /* Enable the USART Interrupt */
-    NVIC_InitStructure.NVIC_IRQChannel = usartIrq;
-    NVIC_InitStructure.NVIC_IRQChannelPreemptionPriority = IRQ_priority;
-    NVIC_InitStructure.NVIC_IRQChannelSubPriority = 0;
-    NVIC_InitStructure.NVIC_IRQChannelCmd = ENABLE;
-    NVIC_Init(&NVIC_InitStructure);
-
-    if (irqType | UART_RX_IRQ)
-        USART_ITConfig(USARTx, USART_IT_RXNE, ENABLE);
-    if (irqType | UART_TX_IRQ)
-        USART_ITConfig(USARTx, USART_IT_TXE, ENABLE);
+        NVIC_InitTypeDef NVIC_InitStructure;
+
+        /* Configure the NVIC Preemption Priority Bits */
+        NVIC_PriorityGroupConfig(NVIC_PriorityGroup_4);
+
+        /* Enable the USART Interrupt */
+        NVIC_InitStructure.NVIC_IRQChannel = usartIrq;
+        NVIC_InitStructure.NVIC_IRQChannelPreemptionPriority = IRQ_priority;
+        NVIC_InitStructure.NVIC_IRQChannelSubPriority = 0;
+        NVIC_InitStructure.NVIC_IRQChannelCmd = ENABLE;
+        NVIC_Init(&NVIC_InitStructure);
+
+        /* Explicitly disable most interrupts.  Only enable ones we want */
+        USART_ITConfig(USARTx, USART_IT_WU, DISABLE);
+        USART_ITConfig(USARTx, USART_IT_CM, DISABLE);
+        USART_ITConfig(USARTx, USART_IT_EOB, DISABLE);
+        USART_ITConfig(USARTx, USART_IT_RTO, DISABLE);
+        USART_ITConfig(USARTx, USART_IT_CTS, DISABLE);
+        USART_ITConfig(USARTx, USART_IT_LBD, DISABLE);
+        USART_ITConfig(USARTx, USART_IT_TC, DISABLE);
+        USART_ITConfig(USARTx, USART_IT_IDLE, DISABLE);
+        USART_ITConfig(USARTx, USART_IT_PE, DISABLE);
+        USART_ITConfig(USARTx, USART_IT_ERR, DISABLE);
```

Comment:
IIRC, these interrupt enablers are set/cleared as a mask.  Might not hurt to define something like UART_IT_ALL with these guys bitwise or'd together and have a single call.